src/storageDataGroup.js

```diff
@@ -360,58 +449,56 @@ class StorageDataGroup {
         }
 
+        let taskId;
         return Promise.resolve()
-            .then(() => new Promise((resolve, reject) => {
-                const req = http.request(opts, (res) => {
-                    const buffer = [];
-                    res.setEncoding('utf8');
-                    res.on('data', (data) => {
-                        buffer.push(data);
-                    });
-                    res.on('end', () => {
-                        let body = buffer.join('');
-                        body = body || '{}';
-                        try {
-                            body = JSON.parse(body);
-                        } catch (e) {
-                            return reject(new Error(`Invalid response object from ${opts.method} to ${opts.path}`));
-                        }
-                        return resolve({
-                            status: res.statusCode,
-                            headers: res.headers,
-                            body
-                        });
-                    });
-                });
-
-                req.on('error', (e) => {
-                    reject(new Error(`${opts.host}:${e.message}`));
-                });
-
-                req.end(JSON.stringify(payload));
-            }))
+            .then(() => {
+                const opts = {
+                    host: 'localhost',
+                    port: 8100,
+                    path: '/mgmt/tm/task/sys/config',
+                    method: 'POST',
+                    headers: {
+                        Authorization: `Basic ${Buffer.from('admin:').toString('base64')}`,
+                        'Content-Type': 'application/json'
+                    }
+                };
+                const payload = {
+                    command: 'save'
+                };
+
+                return iControlRestRequest(opts, payload);
+            })
+            .then((res) => {
+                if (res.status !== 200) {
+                    return Promise.reject(new Error(`failed to submit save sys config task:${JSON.stringify(res.body)}`));
+                }
+                taskId = res.body._taskId;
+                const opts = {
+                    host: 'localhost',
+                    port: 8100,
+                    path: `/mgmt/tm/task/sys/config/${taskId}`,
+                    method: 'PUT',
+                    headers: {
+                        Authorization: `Basic ${Buffer.from('admin:').toString('base64')}`,
+                        'Content-Type': 'application/json'
+                    }
+                };
+                const payload = { _taskState: 'VALIDATING' };
+
+                return iControlRestRequest(opts, payload);
+            })
             .then((response) => {
-                if (response.status !== 200) {
-                    return Promise.reject(new Error(`failed to save sys config:${JSON.stringify(response.body)}`));
+                if (response.status !== 202) {
```

Comment:
Do we always get a 202 here or might we get a 200 if the task finishes right away?

Reply:
According to the documentation:

If the request was successful, you should see a response similar to the following:
{
 "_code": 202,
 "errorStack": [],
 "message": "Task will execute asynchronously."
}